Upgrade to Ruby 3.1

hiromi-mi opened the pull request:
- install webrick as gem
  - webrick was removed from standard library
- `URI.encode` -> `URL.encode_www_form_component`
- sass -> sassc
  - Ruby Sass Has Reached End-of-Life https://github.com/sass/ruby-sass
- update dependency

hiromi-mi committed:
work on ruby 3.1

- install webrick as gem
- URI.encode -> URL.encode_www_form_component
- sass -> sassc
- update dependency

hanazuki commented on Gemfile:
```diff
@@ -1,10 +1,11 @@
 source 'https://rubygems.org'
 
+gem 'webrick'
```
productionでは必要ないので，`group :development`で囲っておいてほしいです

hiromi-mi committed:
Merge remote-tracking branch 'origin/master' into feature/ruby31

hanazuki commented on lib/app.rb:
```diff
@@ -93,7 +93,7 @@ def escape(str)
 
         if revisions
-          diff_url = url(URI.encode("#{path}?diff&from=#{revisions[:old]}&to=#{revisions[:new]}"))
+          diff_url = url(URI.encode_www_form_component("#{path}?diff&from=#{revisions[:old]}&to=#{revisions[:new]}"))
```
これは等価な書き換えではないです
```ruby
URI.encode("path?diff&from=a")  # => "path?diff&from=a"
URI.encode_www_form_component("path?diff&from=a")  # => "path%3Fdiff%26from%3Da" 
```

hiromi-mi committed:
add ruby platform to Gemfile.lock

hiromi-mi committed:
クエリパラメータを URI.encode_www_form でデコード

https://docs.knapsackpro.com/2020/uri-escape-is-obsolete-percent-encoding-your-query-string

hiromi-mi committed:
webrickをdevelopment専用に

hiromi-mi committed:
Merge remote-tracking branch 'origin/master' into feature/ruby31

hanazuki commented:
`URI.encode_www_form_component(path)`のようにしている箇所で`path`に"/"が含まれると"%2F"にエンコードされてしまう点で挙動が変わっていますね．
PATH_INFOに%2Fがきてもページの閲覧はできそうですが，ブラウザのロケーションバーでの表示がかわって見づらくなりそうです

hanazuki committed:
Merge remote-tracking branch 'origin/master' into feature/ruby31
